internal/controllers/machineset/machineset_controller.go

```diff
@@ -1208,6 +1209,20 @@ func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, s *scope) (
 	if isDeploymentChild(ms) {
 		if owner.Annotations[clusterv1.RevisionAnnotation] != ms.Annotations[clusterv1.RevisionAnnotation] {
 			// MachineSet is part of a MachineDeployment but isn't the current revision, no remediations allowed.
+			machinesToRemediate := make([]*clusterv1.Machine, 0, len(filteredMachines))
+			for _, m := range filteredMachines {
+				if conditions.IsFalse(m, clusterv1.MachineOwnerRemediatedCondition) {
```

Comment:
q: should we check also health check succeeded false? (I will do this also below at L1250)

Reply:
Not sure if we want to change the controller behavior. But I can if we want to

Reply:
We also don't have the logic that KCP has to get rid of OwnerRemediated if the MachineHealthCheckSucceeded condition is true

Reply:
What about:
- Implementing the missing logic now
- Adding a TODO for changing the controller behaviour when we move to v1beta2 conditions

End state, we will be "aligned" with what KCP is doing

Reply:
Are these mutually exclusive options or are you suggesting to implement the missing logic without changing the controller behavior? :)

Reply:
Discussed. We'll align to KCP basically